src/main/java/org/opentripplanner/middleware/utils/JsonUtils.java

```diff
@@ -72,6 +76,27 @@ public static void logMessageAndHalt(Request request, int statusCode, String mes
         return null;
     }
 
+    /**
+     * Utility method to parse generic object from HTTP response.
+     */
+    public static <T> T getPOJOFromHttpBody(HttpResponse<String> response, Class<T> clazz) {
+        if (response == null || response.statusCode() >= 400) {
+            String result = response == null ? "bad response!" : response.body();
+            LOG.error("Error found in HTTP response: {}", result);
+            return null;
+        }
```

Comment:
Can you make a util method for this and make both `getPOJOFromHttpBody` and `getPOJOFromHttpBodyAsList` use it?